compiler/modules/front.syntax/astString.bal

```diff
@@ -247,16 +247,33 @@ function typeDescToWords(Word[] w, TypeDesc td, boolean|BinaryTypeOp wrap = fals
         return;
     }
     else if td is MappingTypeDesc {
-        w.push("map", CLING, "<", CLING);
         TypeDesc? rest = td.rest;
-        if rest == () {
-            typeDescToWords(w, { startPos: td.startPos, endPos: td.endPos, builtinTypeName: "never" });
+        if td.fields.length() > 0 {
+            w.push("record", "{|");
+            boolean firstInBlock = true;
+            foreach var f in td.fields {
+                w.push(<Word>(firstInBlock ? LF_INDENT : LF));
+                firstInBlock = false;
+                typeDescToWords(w, f.typeDesc);
+                w.push(f.name, CLING, ";");
+            }
+            if rest != () {
```

Comment:
I don't think the firstInBlock logic is working right with rest. Are we properly handling the case where there is a `rest` field but no `members`?

Reply:
Can we differentiate between `type M map<int>` and
```
type R record {|
    int ...;
|}
```
in MappingTypeDesc ?

Reply:
Those are two ways of writing the same thing. we should pick `map<int>` as the canonical form.

Reply:
in the current implementation second form will also be serialized as `map<int>`